modules/tinymce/src/plugins/lists/test/ts/browser/ToggleListInTableCell.ts

```diff
@@ -0,0 +1,79 @@
+import { Keys } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyAssertions, TinyContentActions, TinyHooks, TinySelections, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/lists/Plugin';
+
+describe('browser.tinymce.plugins.lists.ToggleListWithEmptyLiTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    indent: false,
+    plugins: 'lists table',
+    toolbar: 'bullist',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin ], true);
+
+  const wrapInsideTable = (content: string): string =>
+    `<table style="border-collapse: collapse; width: 100%;" border="1"><colgroup><col style="width: 50%;"><col style="width: 50%;"></colgroup><tbody><tr><td>${content}</td></tr></tbody></table>`;
+
+  const wrapInsideTableWithList = (content: string): string =>
+    `<table style="border-collapse: collapse; width: 100%;" border="1"><colgroup><col style="width: 50%;"><col style="width: 50%;"></colgroup><tbody><tr><td><ul><li>${content}</li></ul></td></tr></tbody></table>`;
+
+  it('TINY-6853: toggle list inside table cell with anchor outside of anchor to the left of the anchor should include the anchor in the list', () => {
+    const editor = hook.editor();
+    const content = '<a href="#">link</a>';
+    editor.setContent(wrapInsideTable(content));
+    TinySelections.setCursor(editor, [ 0, 1, 0, 0, 0 ], 0);
+    // Needed to move the cursor outside of the link
+    TinyContentActions.keydown(editor, Keys.left());
+    TinyContentActions.keyup(editor, Keys.left());
+    TinyUiActions.clickOnToolbar(editor, 'button[aria-label="Bullet list"]');
+    TinyAssertions.assertContent(editor, wrapInsideTableWithList(content));
+  });
+  it('TINY-6853: toggle list inside table cell with anchor outside of anchor to the right of the anchor should include the anchor in the list', () => {
```

Comment:
Maybe some newlines between the it calls just to fluff it up a bit. To me at least that is easier to read but feel free to ignore if you don't agree.